diff --git a/src/index.js b/src/index.js
index 7d9d2337a3363b81bb13b2538feefa99f1d401a2..f669ba78d8ffc34a44421ad73a37e59b3bc700eb 100644
--- a/src/index.js
+++ b/src/index.js
@@ -19,7 +19,7 @@ const loadStore = options => {
 	};
 	if (options.adapter || options.uri) {
 		const adapter = options.adapter || /^[^:+]*/.exec(options.uri)[0];
-		return new (require(adapters[adapter]))(options);
+		return new (({}))(options);
 	}
 
 	return new Map();
